CMakeLists.txt

```diff
@@ -41,6 +41,45 @@ target_include_directories(control_toolbox PUBLIC
 ament_target_dependencies(control_toolbox PUBLIC ${THIS_PACKAGE_INCLUDE_DEPENDS})
 target_compile_definitions(control_toolbox PRIVATE "CONTROL_TOOLBOX_BUILDING_LIBRARY")
 
+
+########################
+# Build control filters
+########################
+find_package(filters REQUIRED)
+find_package(generate_parameter_library REQUIRED)
+find_package(geometry_msgs REQUIRED)
+find_package(pluginlib REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(Eigen3 REQUIRED)
+
+set(CONTROL_FILTERS_INCLUDE_DEPENDS
+  Eigen3
+  generate_parameter_library
+  filters
+  geometry_msgs
+  pluginlib
+  rclcpp
+)
+
+add_library(low_pass_filter SHARED
+  src/control_filters/low_pass_filter.cpp
+)
+
+target_include_directories(low_pass_filter PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<BUILD_INTERFACE:${EIGEN3_INCLUDE_DIR}>
+    $<INSTALL_INTERFACE:include>
+)
+generate_parameter_library(
+  low_pass_filter_parameters
+  src/control_filters/low_pass_filter_parameters.yaml
+)
+target_link_libraries(low_pass_filter low_pass_filter_parameters)
+ament_target_dependencies(low_pass_filter ${CONTROL_FILTERS_INCLUDE_DEPENDS})
```

Comment:
```suggestion
target_compile_features(low_pass_filter PUBLIC cxx_std_17)
target_include_directories(low_pass_filter PUBLIC
    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
    $<BUILD_INTERFACE:${EIGEN3_INCLUDE_DIR}>
    $<INSTALL_INTERFACE:include/control_filters>
)
target_link_libraries(low_pass_filter PUBLIC low_pass_filter_parameters)
ament_target_dependencies(low_pass_filter ${CONTROL_FILTERS_INCLUDE_DEPENDS})
```

Reply:
This line `$<INSTALL_INTERFACE:include/control_filters>`  maybe have to be `$<INSTALL_INTERFACE:include/control_toolbox>` - please test which version is working.

Reply:
i am actually surprised one needs to install the includes in a different folder than the project name or our own name (control_filters).
The code before my PR #152 (which is only 3 weeks old from #145 ) already installs in a control_toolbox sub-folder of the project name, making it look like 
`my_ws/install/control_toolbox/include/control_toolbox/control_toolbox`
`my_ws/install/control_toolbox/include/control_toolbox/control_filters`
is that really what we want ? I have difficulties understanding the "override" thing, so maybe there is a good reason, otherwise I would expect:
`my_ws/install/control_toolbox/include/control_toolbox`
and for the control filters
`my_ws/install/control_toolbox/include/control_filters`

it is possible this got un-noticed, because with underlays still having the control_toolbox includes looking as I would expect it, this simply worked (the underlay libs were found ?) until this new change gets released ?

Reply:
I understand this is confusing. It is also for all of us, but this is how ament und colcon work...

you can read here more details and check references why we have to do that: https://github.com/ros-controls/ros2_control/pull/926

Reply:
> it is possible this got un-noticed, because with underlays still having the control_toolbox includes looking as I would expect it, this simply worked (the underlay libs were found ?) until this new change gets released ?

You are adding here new libs, and they work without problem. The issue comes when you have a lib, install in the system and want to override it as in ROS 1 – this doesn't work without these tricks. After loosing already a dozen of hours debugging non-existing issues, we decided to take this ugly implementation and just make things to be easily used.

Reply:
OK, maybe #145 should have at least pointed to the doc justifying that, then I would not have asked.

This is not nice but very important and something others will stumble upon. I had seen the override warnings in colcon build but I did not relate the "ordering" to the strange naming.

I see that ROS 1 solved that known issue of overlays by having re-ordering of the includes done in the cmake when catkin was called.

I hope this gets solved one day.

Reply:
> This line `$<INSTALL_INTERFACE:include/control_filters>` maybe have to be `$<INSTALL_INTERFACE:include/control_toolbox>` - please test which version is working.

since the destination of the source includes installation is this name-trick to permit override, the install interface of the filters has to be in the same folder as this name-trick. 
if one wants to have a different name-trick applied to the `control_filters` so that it goes in `control_filters/control_filters`
the install cmake command must be modified to do 2 sub-folder installs for `control_toolbox` and `control_filters` separately.

depends what you prefer:

`include/control_toolbox/control_toolbox`
`include/control_toolbox/control_filters`
or
`include/control_toolbox/control_toolbox`
`include/control_filters/control_filters`

Reply:
For my understanding : 
Could you confirm there is no override issue with the generated libraries headers (in case a this generate library yaml gets changed in an overlay), although they get installed in `include/low_pass_filter_parameters/` and not in `include/low_pass_filter_parameters/low_pass_filter_parameters` because the `target_link_library` sees it is a local target and finds the local header  ?

Reply:
We didn't test this yet with generated parameters.